landlab/utils/tests/test_stream_length.py

```diff
@@ -0,0 +1,156 @@
+import numpy as np
+import math
+from nose.tools import assert_equal, assert_true, assert_false, assert_raises
+from numpy.testing import assert_array_equal
+
+from landlab import RasterModelGrid, FieldError, HexModelGrid, CLOSED_BOUNDARY
+from landlab.components import FlowAccumulator, FastscapeEroder, FlowDirectorSteepest
+from landlab.utils.stream_length import calculate_stream_length
+
+
+def test_no_flow_recievers():
+    """Test that correct error is raised when no flow recievers are on the grid."""
+    # instantiate a model grid, do not run flow accumulation on it
+    mg = RasterModelGrid(30, 70)
+    # test that the stream length utility will fail because of a ValueError
+    assert_raises(FieldError, calculate_stream_length, mg)
+
+
+def test_no_upstream_array():
+    """Test that correct error is raised when no flow__upstream_node_order."""
+    # instantiate a model grid, do not run flow accumulation on it
+    mg = RasterModelGrid(30, 70)
+    z = mg.add_ones('node','topographic__elevation')
+    fd = FlowDirectorSteepest(mg)
+    fd.run_one_step()
+    # test that the stream length utility will fail because of a ValueError
+    assert_raises(FieldError, calculate_stream_length, mg)
+
+
+def test_stream_length_regular_grid_d8():
+    """Test to demonstrate that stream_length utility works as expected with regular grids"""
+    # instantiate a model grid
+    mg = RasterModelGrid((5, 4), spacing=(1, 1))
+    # instantiate an elevation array
+    z = np.array ([[0,0,0,0],[0,21,10,0],[0,31,20,0],[0,32,30,0],[0,0,0,0]],dtype='float64')
+    # add the elevation field to the grid
+    mg.add_field('node','topographic__elevation', z)
+    # instantiate the expected flow_length array
+    # considering flow directions calculated with D8 algorithm
+    flow_length_expected = np.array ([[0,0,0,0],[0,1,0,0],[0,math.sqrt(2),1,0],[0,1+math.sqrt(2),2,0],[0,0,0,0]],dtype='float64')
+    flow_length_expected = np.reshape (flow_length_expected,mg.number_of_node_rows*mg.number_of_node_columns)
+    #setting boundary conditions
+    mg.set_closed_boundaries_at_grid_edges(bottom_is_closed=True, left_is_closed=True, right_is_closed=True, top_is_closed=True)
+    # calculating flow directions with FlowAccumulator component
+    fr = FlowAccumulator(mg, flow_director = 'D8')
```

Comment:
Perhaps unexpectedly, no spaces are around equal signs for keyword arguments, although the equal side adjacent to the output (the output here is fr) does have spaces around it:
```python
fr = FlowAccumulator(mg, flow_director='D8')
```